Make build treeshakable

tnagorra opened the pull request:
## Breaking Changes

- Remove language enum
- Change default normalize lang to english

## Non-Breaking Changes

- Pass index on all key selectors
- Make build treeshakable
- Add functions max and min
- Add unimported
- Update babelrc
- Update tsconfig
- Update gitignore
- Update eslintrc
- Generate docs

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] eslint issues
- [x] typescript issues
- [x] test errors

## This PR contains valid:

- [x] tests
- [x] acceptable test coverage
- [x] build works
- [x] docs

tnagorra committed:
Make build treeshakable

- Breaking Changes
    - Remove language enum
    - Change default normalize lang to english

- Non-Breaking Changes
    - Pass index on all key selectors
    - Make build treeshakable
    - Add functions max and min
    - Add unimported
    - Update babelrc
    - Update tsconfig
    - Update gitignore
    - Update eslintrc
    - Generate docs

frozenhelium commented on src/list.ts:
```diff
@@ -211,3 +211,35 @@ export function unique<T>(list: Maybe<T[]>, getItemHash?: ((item: T) => string |
+    if (!list || list.length <= 0) {
+        return undefined;
+    }
+    let maxItem = list[0];
+    let maxValue = comparator(maxItem);
+    list.forEach((item) => {
+        const myValue = comparator(item);
+        if (myValue > maxValue) {
+            maxValue = myValue;
+            maxItem = item;
+        }
+    });
+    return maxItem;
```
Why not use `reduce` here?

frozenhelium replied:
Also check for undefined and null values in the list

frozenhelium commented on src/list.ts:
```diff
@@ -211,3 +211,35 @@ export function unique<T>(list: Maybe<T[]>, getItemHash?: ((item: T) => string |
+    }
+    let minItem = list[0];
+    let minValue = comparator(minItem);
+    list.forEach((item) => {
```
`reduce`

tnagorra committed:
Add stricter type checking

- Add robust checks for min and max functions
- Skip lint and typecheck for tests
